### PR TITLE
nixos/vaultwarden: drop outdated ownership requirements for environmentFile

### PR DESCRIPTION
**Pull request by networkException**
## Description of changes

This pull request introduces stronger wording for using the 
`services.vaultwarden.environmentFile` option when specifying secrets
as well as removes wrongly documented permission requirements for the file.

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: nixos/modules/services/security/vaultwarden/default.nix [hunk 1]
```diff
@@ -131,21 +131,15 @@ in {
         Additional environment file as defined in {manpage}`systemd.exec(5)`.
 
         Secrets like {env}`ADMIN_TOKEN` and {env}`SMTP_PASSWORD`
-        may be passed to the service without adding them to the world-readable Nix store.
+        should be passed to the service without adding them to the world-readable Nix store.
 
         Note that this file needs to be available on the host on which
         `vaultwarden` is running.
```
Comment (SuperSandro2000) [inline nixos/modules/services/security/vaultwarden/default.nix:137]:
```suggestion
        Note that this file needs to be available on the host on which `vaultwarden` is running.
```

File: nixos/modules/services/security/vaultwarden/default.nix [hunk 2]
```diff
@@ -131,21 +131,15 @@ in {
...
 
         As a concrete example, to make the Admin UI available
         (from which new users can be invited initially),
         the secret {env}`ADMIN_TOKEN` needs to be defined as described
```
Comment (SuperSandro2000) [inline nixos/modules/services/security/vaultwarden/default.nix:141]:
```suggestion
        As a concrete example, to make the Admin UI available (from which new users can be invited initially),
        the secret {env}`ADMIN_TOKEN` needs to be defined as described
```

**Review by SuperSandro2000 — COMMENTED** [review]
we might as well not limit ourselves to such short lines to make the sentences more readable.

**Comment by SuperSandro2000** [comment]
Those where from back when the service sourced the envs itself and not systemd.

**Comment by networkException** [comment]
Hm, I get that its no problem for you to understand the intention of my changes with all the context a frequent nixpkgs contributor has.

I believe looking at a change alone can be daunting for someone not familiar with systemd or the issue with having secrets in the /nix/store (and why things end up there) however, as such I try to state intention for changes explicitly.

Perhaps next time thats something for you to consider as well, given that there's no harm in a somewhat longer commit message :)

**Pull request merged**